components/brave_origin/policy/brave_origin_policy_provider.cc

```diff
@@ -0,0 +1,261 @@
+void BraveOriginPolicyProvider::Init(policy::SchemaRegistry* registry) {
+  // Call parent class Init first
+  policy::ConfigurationPolicyProvider::Init(registry);
+
+  // Check which policies are already set by external providers (admin policies)
+  // so we don't override them
+  CheckExternallyManagedPolicies();
+
+  // Trigger immediate policy loading to ensure policies are available in
+  // chrome://policy
+  RefreshPolicies(policy::PolicyFetchReason::kBrowserStart);
+}
+
+void BraveOriginPolicyProvider::Shutdown() {
+  // Call parent class Shutdown
+  policy::ConfigurationPolicyProvider::Shutdown();
+}
+
+void BraveOriginPolicyProvider::RefreshPolicies(
+    policy::PolicyFetchReason reason) {
+  policy::PolicyBundle bundle = LoadPolicies();
+
+  // Mark as loaded after successfully loading policies (or empty bundle if
+  // user is not a BraveOrigin user, which is also a valid state)
+  first_policies_loaded_ = true;
+
+  UpdatePolicy(std::move(bundle));
+}
+
+bool BraveOriginPolicyProvider::IsFirstPolicyLoadComplete(
+    policy::PolicyDomain domain) const {
+  return first_policies_loaded_;
+}
+
+policy::PolicyBundle BraveOriginPolicyProvider::LoadPolicies() {
+  policy::PolicyBundle bundle;
+
+  // Check if BraveOrigin feature is enabled
+  bool is_brave_origin_user = IsBraveOriginEnabled();
+
+  if (!is_brave_origin_user) {
+    return bundle;
+  }
+
+  // Create policy map for Chrome domain
+  policy::PolicyMap& policy_map = bundle.Get(
+      policy::PolicyNamespace(policy::POLICY_DOMAIN_CHROME, std::string()));
+
+  // Set BraveOrigin policies using mappings from BraveOriginService
+  const auto policy_mappings = BraveOriginService::GetPolicyKeyMappings();
+
+  if (!local_state_) {
+    return bundle;
+  }
+
+  for (const auto& [policy_key, pref_name] : policy_mappings) {
+    const BraveOriginPrefInfo* pref_info =
+        BraveOriginService::GetPrefInfo(pref_name);
+    if (!pref_info) {
+      continue;
+    }
+    if (pref_info->scope == BraveOriginPolicyScope::kProfile) {
+      SetBraveOriginPolicyForPref(policy_map, policy_key, pref_name,
+                                  local_state_);
+    } else if (pref_info->scope == BraveOriginPolicyScope::kGlobal) {
+      SetBraveOriginGlobalPolicyForPref(policy_map, policy_key, pref_name,
+                                        local_state_);
+    } else {
+      continue;
+    }
+  }
+
+  return bundle;
+}
+
+bool BraveOriginPolicyProvider::IsPolicySetByExternalProvider(
+    const std::string& policy_key) const {
+  if (!local_state_) {
+    return false;
+  }
+
+  // Get the pref name for this policy key
+  const auto policy_mappings = BraveOriginService::GetPolicyKeyMappings();
+  auto it = policy_mappings.find(policy_key);
+  if (it == policy_mappings.end()) {
+    return false;
+  }
+
+  const std::string& pref_name = it->second;
+
+  // Build the externally managed key
+  std::string externally_managed_key =
+      pref_name + brave_origin::prefs::kExternallyManagedSuffix;
```

Comment:
If we do decide to keep storing the "externally managed" state in prefs, perhaps we should carve out another dict or list pref explicitly for these keys. That could eliminate the need for the suffix.